phabricatoremails/render/mailbatch.py

```diff
@@ -0,0 +1,112 @@
+from dataclasses import dataclass
+from typing import Optional, List, Dict
+
+from phabricatoremails.mail import OutgoingEmail
+from phabricatoremails.render.events.common import Recipient
+from phabricatoremails.render.events.phabricator import Revision
+from phabricatoremails.render.events.phabricator_secure import SecureRevision
+from phabricatoremails.render.template import TemplateStore
+
+
+class MailBatch:
+    """Creates several outgoing emails from a single Phabricator event
+
+    Most Phabricator events trigger multiple emails from the same source data.
+    For example, if a reviewer accepts a revision, then the author and all the
+    other reviewers are notified.
+
+    Different recipients may receive different emails from different templates, but
+    they're all parameterized by the same event.
+    """
+
+    def __init__(self, css_inline, template_store: TemplateStore):
+        self._targets = []
+        self._css_inline = css_inline
+        self._template_store = template_store
+
+    def target(self, recipient: Optional[Recipient], template_name: str, **kwargs):
```

Comment:
It's worth considering the context in which `phabricator-emails` is used. It isn't a module that's being publicized, and there's existing usages of these methods.

I agree that this one can be more clear because of the mysterious `kwargs` parameter, so I'll add some extra documentation to `target()` and `target_all()`. However, the other methods have helpful type signatures, and I'm not sure that documenting every class and method in a project is the most efficient use of our time due to diminishing returns.

Reply:
I promise that I'm familiar with the idea of amortization, a new car seems cheap if you just pay a little bit every month.

It's possible to over-document, which at worse leads to bad/out-of-date documentation, and at best is just extra verbosity.

I agree that documentation is useful, but adding docstrings to every method is excessive. This isn't a library that's going to be used externally with no context - these classes and methods are all internal code.

Reply:
> I agree that documentation is useful, but adding docstrings to every method is excessive. This isn't a library that's going to be used externally with no context - these classes and methods are all internal code.

I would completely disagree with this; adding docstrings to as many methods as possible is not excessive, it is good practice. If it's possible to put things in a docstring instead of elsewhere in a separate readme, they should go in the docstring.

Reply:
> If it's possible to put things in a docstring instead of elsewhere in a separate readme, they should go in the docstring.

I agree, if information is scoped enough to be written at the point of implementation, then they make more sense there than the readme.

> I would completely disagree with this; adding docstrings to as many methods as possible is not excessive, it is good practice.

Hmm, do you agree that documentation has a maintenance cost?